packages/core/src/typegenTypes.ts

```diff
@@ -165,7 +165,7 @@ type AllowAllEvents = {
 };
 
 export type ResolveTypegenMeta<
-  TTypesMeta extends TypegenConstraint,
+  TTypesMeta,
```

Comment:
`ResolveTypegenMeta` isn't supposed to be used publicly, so we can just trust its input. The problem was that after this `infer` trick above TS couldn't provide that the `EvaluatedTypesMeta` actually satisfies this constraint. An alternative approach to removing this generic would be to use `Cast<EvaluatedTypesMeta, TypegenConstraint>` at those "call sites". This seemed to be easier though.

Simplified demo of the problem:
[TS playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoE8AOEDmEQDCA9iAM5hRyhjIDeAUMsgEQACrAHuXJAPRhZc+ZgC5kAIyJEANhDggA3PQC+9egOzIAShFIyAbtAA8qZBA6QQAE1JpBeQiXKVqAPmQBeNEvpWICaTgoFBgAVxAEMGASZEQkTDAAJhMzC3wbO2wHYjIKKnBXAApGWKgcMVNzSwzQGGhkAFF9OGlQnggrZAB+bV0DYyaWtsgrdzEQCEMoegBKMX0iYCslIA)